### PR TITLE
ref(ui) Refactor team select implementation to be the same

### PR DESCRIPTION
**Pull request by markstory**
Collapse the implementation of the two team select widgets to use the same implementation.

Fixes SEN-242

### CODE REVIEW

File: src/sentry/static/sentry/app/views/settings/project/projectTeams.jsx [hunk 1]
```diff
@@ -126,11 +65,12 @@ class ProjectTeams extends AsyncView {
     });
   };
 
-  handleAdd = selection => {
-    if (this.state.loading) return;
-
-    const team = this.state.allTeams.find(tm => tm.id === selection.value);
+  handleAdd = teamSlug => {
+    if (this.state.loading) {
```
Comment (billyvg) [inline src/sentry/static/sentry/app/views/settings/project/projectTeams.jsx]: imo we should enforce `{}` in linter

File: src/sentry/static/sentry/app/views/settings/project/projectTeams.jsx [hunk 2]
```diff
@@ -108,6 +29,24 @@ class ProjectTeams extends AsyncView {
...
+    }
+
+    const {orgId, projectId} = this.props.params;
+    const team = this.state.allTeams.find(tm => tm.slug === teamSlug);
+
+    removeTeamFromProject(this.api, orgId, projectId, team.slug)
+      .then(() => this.handleRemovedTeam(team))
+      .catch(() => {
+        this.setState({
+          error: true,
```
Comment (billyvg) [inline src/sentry/static/sentry/app/views/settings/project/projectTeams.jsx]: Do we want to put the entire view in an error state in this case? Would an error toast (`addErrorMessage`) be better?

**Review by billyvg — APPROVED** [review]
Yay for reusing code

File: src/sentry/static/sentry/app/views/settings/project/projectTeams.jsx (hunk 1 above)
Comment (markstory) [reply src/sentry/static/sentry/app/views/settings/project/projectTeams.jsx]: I would love that. I'll look into enabling those rules.

**Pull request merged**